### PR TITLE
Differentiable Nemov Gamma_c fast-ion confinement proxy

### PR DESCRIPTION
**Pull request by rogeriojorge**
Implements the differentiable Nemov Gamma_c fast-ion confinement proxy (plan Phase 21 / 31.1) — the last major physics-objective gap against DESC's library — on the existing bounce kernel and field-line machinery, exported from `vmex.optimize` as `GammaC` alongside its neighbors.

## Formula

```
gamma_c = (2/pi) arctan(v_r / v_p)
Gamma_c = pi/(4 sqrt 2) < int_{1/B_max}^{1/B} dlambda B/sqrt(1 - lambda B) gamma_c^2 >
```

with `v_r`, `v_p` the bounce-averaged radial and poloidal-tangential magnetic drifts and `<.>` the flux-surface average — Nemov, Kasilov, Kernbichler, Leitold, Phys. Plasmas 15, 052501 (2008), eq. 61, organized as in Velasco et al., Nucl. Fusion 61, 116059 (2021), eqs. 14–16 (expanded over wells as `pi/(8 sqrt 2) int dlambda sum_w v tau_b gamma_c^2 / int dl/B`). The arctan argument uses **Nemov's transit-stable form** — DESC's rewrite of Nemov's eqs. 21–22 into single-valued periodic maps (`desc.compute._fast_ion`, on the bounce-averaging kernel of Unalmis et al., JPP 92(3) 2026, doi:10.1017/S0022377826101652):

```
tan(pi gamma_c / 2) = int dl w (BxgradB).grad psi / B^3
                      -------------------------------------------------------------
                      (|grad rho| ||e_alpha||)_{Bmin} int dl [w dB/drho|_PEST + sqrt(1-lambda B) K] / B
w = (1 - lambda B/2)/sqrt(1 - lambda B),  K = iota_rho (grad psi x b).grad phi - 2 dB/drho|_PEST + B dB^phi/drho|_PEST / B^phi
```

The literal Velasco line-integral form was implemented first and rejected on measurement: its poloidal drift carries a secular shear term growing with the sampled line length (the caveat DESC records on its own Velasco variant), which produced a physically inverted QA-vs-CTH ordering. All geometric ingredients are exact spectral point evaluations on PEST field lines reusing `vmex.core.stability`'s machinery; the singular bounce integrals reuse `vmex.core.bounce.bounce_action`, minimally extended in place with the Nemov drift/parallel/softmin kernels.

## Validation

**Drift-kinetic identity FD tests** (li383, s = 0.8, per well, self-contained):
- `d_alpha J` vs `2 psi_edge int w (BxgradB).grad s/B^3 dl`: 0.1088 vs 0.1117 (2.6%)
- well-following `-(1/2) dJ/drho` vs the Nemov drift2 combination: 0.8765 vs 0.8567 (2.3%)
- `dJ/ds` in the fixed-label gauge vs the secular-inclusive `v_M.grad alpha` integral (Velasco eq. 10): −1.003 vs −1.097
- `||e_alpha||` vs position FD: exact to 7 digits

**Wout-table referee** for the disputed ingredient: mean `|dB/drho|` at fixed PEST angles on li383 (ns = 49): wout FD 0.3354 / 0.1823 (s = 0.79 / 0.25) vs vmex 0.335 / 0.182 (exact) vs DESC-loaded 0.795 / 0.207 — DESC's wout refit inflates outer-surface radial derivatives, which is why parity is asserted at the inner surface only.

**DESC parity** (same vmex-written li383 wout, ns = 49, `VMECIO.load` + `eq.compute("Gamma_c")`): s = 0.25: vmex 3.585e-2 vs DESC 3.688e-2 (**2.8%**; DESC's independent 1D `old Gamma_c` gives 4.63e-2/1.85e-2/1.65e-2 across s = 0.27/0.53/0.79, consistent with its own 2D values).

**Physics anchors** (nalpha 7, 3 poloidal transits, 64 pts/transit, 24 pitch, s = 0.5):
- circular tokamak: **9.8e-7** (exact axisymmetric limit `d_alpha J = 0`; pure quadrature noise)
- LandremanPaul2021 QA (ns 13): **3.8e-3** (1.1e-3 at doubled sampling — near-omnigenous values are budget-limited upper bounds)
- li383 / NCSX: **4.11e-2** (4.20e-2 at doubled sampling, stable to 2%)
- mirror-ripple ray on the QA boundary: Gamma_c x4.4 up, NEO_JAX eps_eff^(3/2) 8.9e-9 → 2.3e-4 (directional consistency with an independent transport metric; documented as exactly that)

**Gradient liveness** (li383, ns 13, implicit `im.run` + `jax.grad`, d/d rbc[n=0,m=1]):
- base sampling: value 2.132e-3, gradient **−0.5413**; central FD −0.4269 (step 1e-4), −0.3850 (step 2e-5) → sign agreement, ratio 1.41 (asserted at 3)
- one refinement step (13/4/96/48): **−0.1186** — sign stable, magnitude not: superbanana layers (tangential-drift zeros where gamma_c → ±1, verified on the offending pitch node) put ~4.6x sampling scatter on the gradient magnitude at PR-lane budgets, while the value converges. Recorded convergence ladder (value / gradient): 1.66e-3/−0.32, 1.83e-3/−0.45, 1.77e-3/−0.13, 1.81e-3/−0.49, 1.90e-3/−0.27 across (7,3,64,24)→(21,6,128,96). The tests assert what is measured — exact-per-discretization gradients, FD-consistent, sign-stable — and the docstrings state the scatter rather than hiding it behind tolerances. Practical guidance (docs): optimize at one fixed resolution.

**Measured design decisions**: pitch quadrature is open-midpoint uniform in the reflecting level (the Unalmis sampling guidance; plain Gauss–Legendre in lambda gave resolution-unstable gradients), wells fade across bounded-trace ends under a half-transit taper applied consistently to the normalizing `int dl/B` (removes end-of-trace blinking, unbiased), and the softmin tangency temperature 0.01 was set by the DESC parity scan (0.05 → 23% bias, 0.01 → 2.8%, 0.005 → 0.2%). A merge-topology gate and a bounce-time floor were prototyped and **removed**: the gate's bias grew with refinement (1.51→1.17e-3 against the ungated 1.66→1.90e-3) and the floor biased values 20–40% without curing the superbanana scatter.

## Local verification (Actions is out of minutes)

- `python3 -m ruff check vmex tests` — clean
- `python3 -m mypy vmex` — clean (68 files)
- `python3 -m pytest tests/test_bounce.py tests/test_gammac.py tests/test_test_manifest.py -q -n 2` — 21 passed, **111 s wall** (test_gammac alone: 129 s serial)
- `python3 -m pytest tests/test_maxj.py tests/test_qi_residuals.py tests/test_qi_maxj_composition.py -q -n 2` — 22 passed, 2 skipped (bounce-kernel consumers unchanged)
- new test node ids registered in `tests/manifest.json` (record row + `pr-physics-field` selector), ownership gate passing

Docs: physics entry in `docs/explanation/confinement.rst`, catalog + gradient-table row in `docs/reference/objectives.rst`, `vmex.core.gammac` in the API listing.

### CODE REVIEW

**Comment by rogeriojorge** [comment]
Local CI matrix (Actions minutes exhausted), branch current with main: quality x5, fast, all 9 physics selectors, device, mirror-coverage>=90 — TOTALS pass=16 fail=0. Independently spot-checked: tests/test_gammac.py 4 passed (87 s). The gradient-magnitude sampling scatter stands as a documented limitation; the objective is used at fixed resolution, as the docstrings state.
